config/rbac/extra-roles.yaml

```diff
@@ -50,3 +50,49 @@ subjects:
     name: external-dns-operator
     namespace: external-dns-operator
 
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: external-dns
+rules:
+  - apiGroups:
+      - networking.k8s.io
+    resources:
+      - ingresses
+    verbs:
+      - get
+      - list
+      - watch
+  - apiGroups:
+      - ""
+    resources:
+      - endpoints
+      - services
+      - pods
+      - nodes
+    verbs:
+      - get
+      - list
+      - watch
+  - apiGroups:
+      - route.openshift.io
+    resources:
+      - routes
+    verbs:
+      - get
+      - watch
+      - list
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: external-dns
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: external-dns
+subjects:
+  - kind: Group
```

Comment:
Can't we use kind: ServiceAccount rather than using Group ?

https://kubernetes.io/docs/reference/access-authn-authz/rbac/#role-binding-examples

Reply:
Miheer, in the examples you linked, I see the following:
> For all service accounts in the "dev" group in the "development" namespace:
> 
> ```yaml
> subjects:
> - kind: Group
>   name: system:serviceaccounts:dev
>   apiGroup: rbac.authorization.k8s.io
>   namespace: development

The only example that uses `kind: ServiceAccount` is for specifying a single ServiceAccount:

> For the default service account in the "kube-system" namespace:
> 
> ```yaml
> subjects:
> - kind: ServiceAccount
>   name: default
>   namespace: kube-system

Reply:
@Miciah 

subjects:
- kind: ServiceAccount
  name: system:serviceaccounts:external-dns
  namespace: external-dns

@Miciah  I was thinking something like this. Why do we want all the service accounts in operands ns to get bind to a cluster role ?

Reply:
We want all serviceaccounts in the operand's namespace to be bound to the clusterrole because that is the only way we can avoid giving the operator broad permissions to create and delete clusterrolebindings.